CoreFoundation/String.subproj/CFString.c

```diff
@@ -1264,12 +1264,16 @@ CF_PRIVATE CFStringRef __CFStringCreateImmutableFunnel3(
 	contentsDeallocator = __CFGetDefaultAllocator();
     }
 
+    #ifndef __ANDROID__
+    // https://bugs.swift.org/browse/SR-2587
+    // crashes String.swift, line 51 when creating String from zero length Data
```

Comment:
IIRC bionic's malloc is built with error checking enabled such that free is a bit more pedantic on heap corruption (and has some gnarly behavioral differences) 

It would be nice to see a backtrace and values of locals at the crash point in the bug, not everyone who might be versed to fix this has easy access to an Android device and development environment for them ;)